### PR TITLE
[CI] Migrate to gcloud Artifact Registry

### PR DESCRIPTION
**Pull request by kuhar**
Container Registries have been deprecated:
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr,
so switch to the Artifact Registry.

The biggest difference for us is that AR allows for
multiple repositories per project. And unline CR, it is supported by
the gcloud python API.

I copied all historical images over and will do a cleanup once all CI
workflows have been migrated.

### CODE REVIEW

**Comment by kuhar** [comment]
Sample successful `docker pull` on my fork: https://github.com/kuhar/llpc/runs/7290386561?check_suite_focus=true

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6b1155ddf81d40d93b3af568ca6bc56e99f8cf9e
<details><summary>CTS tests (Failed: 2/179656)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35418/63738 (55.6%)</li>
<li><details><summary>Failed: 1/63738 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined
Stack trace: Script:
Fail

```
</p></details></li>

<li>Not Supported: 28319/63738 (44.4%)</li><li>Warnings: 0/63738 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27529/52180 (52.8%)</li>
<li>Failed: 0/52180 (0.0%)</li>
<li>Not Supported: 24651/52180 (47.2%)</li><li>Warnings: 0/52180 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37583/63738 (59.0%)</li>
<li><details><summary>Failed: 1/63738 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined
Stack trace: Script:
Fail

```
</p></details></li>

<li>Not Supported: 26154/63738 (41.0%)</li><li>Warnings: 0/63738 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
retest this please

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6b1155ddf81d40d93b3af568ca6bc56e99f8cf9e
<details><summary>CTS tests (Failed: 2/179656)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35418/63738 (55.6%)</li>
<li><details><summary>Failed: 1/63738 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined
Stack trace: Script:
Fail

```
</p></details></li>

<li>Not Supported: 28319/63738 (44.4%)</li><li>Warnings: 0/63738 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27529/52180 (52.8%)</li>
<li>Failed: 0/52180 (0.0%)</li>
<li>Not Supported: 24651/52180 (47.2%)</li><li>Warnings: 0/52180 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37583/63738 (59.0%)</li>
<li><details><summary>Failed: 1/63738 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined
Stack trace: Script:
Fail

```
</p></details></li>

<li>Not Supported: 26154/63738 (41.0%)</li><li>Warnings: 0/63738 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by Flakebi** [comment]
@amdrexu, can we exclude the dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined test? It relies on undefined behavior in SPIR-V ([Khronos issue](https://gitlab.khronos.org/Tracker/vk-gl-cts/-/issues/3787)) and llvm recently started to optimize that, which is why the test fails now.

**Comment by Flakebi** [comment]
Sounds good to me if it works. Can you also update the example commands in https://github.com/GPUOpen-Drivers/llpc/blob/05cdb2a1c9ca67d9dcc70d4d68652601c870ef8f/docker/llpc.Dockerfile#L5 and https://github.com/GPUOpen-Drivers/llpc/blob/05cdb2a1c9ca67d9dcc70d4d68652601c870ef8f/docker/llpc-clang-tidy.Dockerfile#L5?

**Comment by amdrexu** [comment]
> @amdrexu, can we exclude the dEQP-VK.graphicsfuzz.cov-instruction-simplify-atanh-log-undefined test? It relies on undefined behavior in SPIR-V ([Khronos issue](https://gitlab.khronos.org/Tracker/vk-gl-cts/-/issues/3787)) and llvm recently started to optimize that, which is why the test fails now.

I think so. Need @jinjianrong and @qiaojbao to assist this.

**Comment by kuhar** [comment]
Updated the examples in dockerfile comments. Thanks, @Flakebi!

**Review by vettoreldaniele — DISMISSED** [review]

**Comment by kuhar** [comment]
Fixed a typo in the clang tidy dockerfile

**Comment by amdvlk-admin** [comment]
### Test summary for commit a33752fc9a2460e388a66df0f0d64f9aab2a77e9
<details><summary>CTS tests (Failed: 0/179654)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 35418/63737 (55.6%)</li>
<li>Failed: 0/63737 (0.0%)</li>
<li>Not Supported: 28319/63737 (44.4%)</li><li>Warnings: 0/63737 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 27529/52180 (52.8%)</li>
<li>Failed: 0/52180 (0.0%)</li>
<li>Not Supported: 24651/52180 (47.2%)</li><li>Warnings: 0/52180 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 37679/63737 (59.1%)</li>
<li>Failed: 0/63737 (0.0%)</li>
<li>Not Supported: 26058/63737 (40.9%)</li><li>Warnings: 0/63737 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by kuhar** [comment]
This needs one more approval from llpcapprovers: @amdrexu @dstutt @trenouf

**Review by amdrexu — APPROVED** [review]

**Comment by kuhar** [comment]
FYI, the CI should recover once the next nightly build passes. But separately, it seems like the previously merged PR causes some lgc test failures: https://github.com/GPUOpen-Drivers/llpc/runs/7314487224?check_suite_focus=true

**Comment by kuhar** [comment]
Update: I manually copied newer docker images to the artifact registry so that we don't have to wait for the next nightly workflow to pass. Nightly amdvlk build is currently blocked by a failing lgc test, @amdrexu @jiaolu: https://github.com/GPUOpen-Drivers/llpc/runs/7314651992?check_suite_focus=true

**Comment by amdrexu** [comment]
Thank you for reporting this. Lu will take a look today and make a fix soon.

**Comment by jiaolu** [comment]
https://github.com/GPUOpen-Drivers/llpc/pull/1883

**Pull request merged**
